### PR TITLE
Fix Dev Build Container

### LINKED ISSUE #180: Docker Dev Target Build Breaks
Running `docker-compose up` fails off of develop on an M1 Mac. 

```docker-compose up
Creating network "nachomemes_default" with the default driver
Building bot
[+] Building 34.5s (14/15)                                                                                                                                             
 => [internal] load build definition from Dockerfile                                                                                                              0.0s
 => => transferring dockerfile: 1.17kB                                                                                                                            0.0s
 => [internal] load .dockerignore                                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                                   0.0s
 => [internal] load metadata for docker.io/library/python:3.10.1-slim                                                                                             0.8s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                                     0.0s
 => [internal] load build context                                                                                                                                 6.3s
 => => transferring context: 244.39MB                                                                                                                             6.2s
 => CACHED [poetry 1/2] FROM docker.io/library/python:3.10.1-slim@sha256:dd3016f846b8f88d8f6c28b43f1da899f07259121aff403091e6f89a703c3d36                         0.0s
 => CACHED [poetry 2/2] RUN python -m pip install --no-cache-dir "poetry==1.1.5" --user                                                                           0.0s
 => CACHED [build 1/3] WORKDIR /app                                                                                                                               0.0s
 => CACHED [build 2/3] COPY poetry.lock pyproject.toml /app/                                                                                                      0.0s
 => CACHED [build 3/3] RUN python -m poetry export --dev -f requirements.txt --output requirements-dev.txt &&   python -m poetry export -f requirements.txt --ou  0.0s
 => [dev 2/6] COPY --from=build /app/requirements-dev.txt /app/requirements.txt                                                                                   0.0s
 => [dev 3/6] RUN apt-get update && apt-get install gcc=4:10.2.1-1 -y                                                                                             9.7s
 => [dev 4/6] WORKDIR /app                                                                                                                                        0.0s
 => ERROR [dev 5/6] RUN python -m pip install --no-cache-dir -r requirements.txt                                                                                 17.5s
------                                                                                                                                                                 
 > [dev 5/6] RUN python -m pip install --no-cache-dir -r requirements.txt:                                                                                             
#14 1.026 Ignoring appnope: markers 'sys_platform == "darwin" and python_version >= "3.7"' don't match your environment                                                
#14 1.026 Ignoring colorama: markers 'sys_platform == "win32" and python_version >= "3.7" and python_full_version >= "3.6.2" and (python_version >= "3.6" and python_full_version < "3.0.0" and platform_system == "Windows" or platform_system == "Windows" and python_version >= "3.6" and python_full_version >= "3.5.0")' don't match your environment
#14 1.028 Ignoring typing-extensions: markers 'python_version < "3.10" and python_full_version >= "3.6.2" and python_version >= "3.6" and python_full_version < "4.0.0"' don't match your environment
#14 1.371 Collecting aiohttp==3.7.4.post0
#14 1.508   Downloading aiohttp-3.7.4.post0.tar.gz (1.1 MB)
#14 1.741   Installing build dependencies: started
#14 3.153   Installing build dependencies: finished with status 'done'
#14 3.155   Getting requirements to build wheel: started
#14 3.270   Getting requirements to build wheel: finished with status 'done'
#14 3.271     Preparing wheel metadata: started
#14 3.390     Preparing wheel metadata: finished with status 'done'
#14 3.408 Collecting argparse==1.4.0
#14 3.424   Downloading argparse-1.4.0-py2.py3-none-any.whl (23 kB)
#14 3.465 Collecting astroid==2.9.2
#14 3.480   Downloading astroid-2.9.2-py3-none-any.whl (254 kB)
#14 3.521 Collecting async-timeout==3.0.1
#14 3.534   Downloading async_timeout-3.0.1-py3-none-any.whl (8.2 kB)
#14 3.550 Collecting asyncio==3.4.3
#14 3.565   Downloading asyncio-3.4.3-py3-none-any.whl (101 kB)
#14 3.593 Collecting attrs==21.2.0
#14 3.610   Downloading attrs-21.2.0-py2.py3-none-any.whl (53 kB)
#14 3.633 Collecting autopep8==1.6.0
#14 3.648   Downloading autopep8-1.6.0-py2.py3-none-any.whl (45 kB)
#14 3.667 Collecting backcall==0.2.0
#14 3.682   Downloading backcall-0.2.0-py2.py3-none-any.whl (11 kB)
#14 3.905 Collecting boto3==1.20.29
#14 3.938   Downloading boto3-1.20.29-py3-none-any.whl (131 kB)
#14 4.241 Collecting botocore==1.23.29
#14 4.260   Downloading botocore-1.23.29-py3-none-any.whl (8.5 MB)
#14 4.685 Collecting chardet==4.0.0
#14 4.701   Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
#14 4.730 Collecting click==8.0.0
#14 4.746   Downloading click-8.0.0-py3-none-any.whl (96 kB)
#14 4.772 Collecting dacite==1.6.0
#14 4.787   Downloading dacite-1.6.0-py3-none-any.whl (12 kB)
#14 4.806 Collecting decorator==5.0.7
#14 4.826   Downloading decorator-5.0.7-py3-none-any.whl (8.8 kB)
#14 4.869 Collecting discord-ext-typed-commands==1.0.3
#14 4.956   Downloading discord_ext_typed_commands-1.0.3-py3-none-any.whl (4.8 kB)
#14 4.998 Collecting discord.py-stubs==1.7.1
#14 5.089   Downloading discord.py_stubs-1.7.1-py3-none-any.whl (63 kB)
#14 5.124 Collecting discord.py==1.7.2
#14 5.144   Downloading discord.py-1.7.2-py3-none-any.whl (786 kB)
#14 5.194 Collecting flask==2.0.2
#14 5.215   Downloading Flask-2.0.2-py3-none-any.whl (95 kB)
#14 5.255 Collecting fuzzywuzzy==0.18.0
#14 5.279   Downloading fuzzywuzzy-0.18.0-py2.py3-none-any.whl (18 kB)
#14 5.293 Collecting idna==3.1
#14 5.310   Downloading idna-3.1-py3-none-any.whl (58 kB)
#14 5.352 Collecting ipython-genutils==0.2.0
#14 5.371   Downloading ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)
#14 5.450 Collecting ipython==7.31.0
#14 5.471   Downloading ipython-7.31.0-py3-none-any.whl (792 kB)
#14 5.561 Collecting isort==5.8.0
#14 5.577   Downloading isort-5.8.0-py3-none-any.whl (103 kB)
#14 5.636 Collecting itsdangerous==2.0.0
#14 5.653   Downloading itsdangerous-2.0.0-py3-none-any.whl (18 kB)
#14 5.679 Collecting jedi==0.18.0
#14 5.698   Downloading jedi-0.18.0-py2.py3-none-any.whl (1.4 MB)
#14 5.788 Collecting jinja2==3.0.0
#14 5.808   Downloading Jinja2-3.0.0-py3-none-any.whl (133 kB)
#14 5.853 Collecting jmespath==0.10.0
#14 5.875   Downloading jmespath-0.10.0-py2.py3-none-any.whl (24 kB)
#14 5.943 Collecting lazy-object-proxy==1.6.0
#14 5.974   Downloading lazy-object-proxy-1.6.0.tar.gz (44 kB)
#14 6.034   Installing build dependencies: started
#14 7.539   Installing build dependencies: finished with status 'done'
#14 7.542   Getting requirements to build wheel: started
#14 7.627   Getting requirements to build wheel: finished with status 'done'
#14 7.628     Preparing wheel metadata: started
#14 7.742     Preparing wheel metadata: finished with status 'done'
#14 7.789 Collecting markupsafe==2.0.0
#14 7.883   Downloading MarkupSafe-2.0.0.tar.gz (18 kB)
#14 8.169 Collecting matplotlib-inline==0.1.2
#14 8.189   Downloading matplotlib_inline-0.1.2-py3-none-any.whl (8.2 kB)
#14 8.207 Collecting mccabe==0.6.1
#14 8.223   Downloading mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
#14 8.457 Collecting multidict==5.1.0
#14 8.472   Downloading multidict-5.1.0.tar.gz (53 kB)
#14 8.547   Installing build dependencies: started
#14 9.878   Installing build dependencies: finished with status 'done'
#14 9.882   Getting requirements to build wheel: started
#14 9.989   Getting requirements to build wheel: finished with status 'done'
#14 9.991     Preparing wheel metadata: started
#14 10.10     Preparing wheel metadata: finished with status 'done'
#14 10.14 Collecting mypy-extensions==0.4.3
#14 10.16   Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
#14 10.23 Collecting mypy==0.930
#14 10.25   Downloading mypy-0.930-py3-none-any.whl (2.4 MB)
#14 10.35 Collecting parso==0.8.2
#14 10.37   Downloading parso-0.8.2-py2.py3-none-any.whl (94 kB)
#14 10.39 Collecting pathspec==0.8.1
#14 10.40   Downloading pathspec-0.8.1-py2.py3-none-any.whl (28 kB)
#14 10.42 Collecting pexpect==4.8.0
#14 10.44   Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
#14 10.46 Collecting pickleshare==0.7.5
#14 10.48   Downloading pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
#14 10.67 Collecting pillow==9.0.0
#14 10.82   Downloading Pillow-9.0.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (4.2 MB)
#14 10.95 Collecting platformdirs==2.4.0
#14 10.97   Downloading platformdirs-2.4.0-py3-none-any.whl (14 kB)
#14 11.01 Collecting prompt-toolkit==3.0.18
#14 11.04   Downloading prompt_toolkit-3.0.18-py3-none-any.whl (367 kB)
#14 11.15 Collecting psutil==5.9.0
#14 11.17   Downloading psutil-5.9.0.tar.gz (478 kB)
#14 11.42 Collecting ptyprocess==0.7.0
#14 11.44   Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
#14 11.46 Collecting pycodestyle==2.8.0
#14 11.49   Downloading pycodestyle-2.8.0-py2.py3-none-any.whl (42 kB)
#14 11.55 Collecting pygments==2.9.0
#14 11.57   Downloading Pygments-2.9.0-py3-none-any.whl (1.0 MB)
#14 11.66 Collecting pylint==2.12.2
#14 11.67   Downloading pylint-2.12.2-py3-none-any.whl (414 kB)
#14 11.71 Collecting python-dateutil==2.8.1
#14 11.73   Downloading python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
#14 11.78 Collecting python-levenshtein-wheels==0.13.2
#14 11.87   Downloading python-Levenshtein-wheels-0.13.2.tar.gz (38 kB)
#14 12.03 Collecting pyyaml==5.4.1
#14 12.05   Downloading PyYAML-5.4.1.tar.gz (175 kB)
#14 12.18   Installing build dependencies: started
#14 14.11   Installing build dependencies: finished with status 'done'
#14 14.12   Getting requirements to build wheel: started
#14 14.58   Getting requirements to build wheel: finished with status 'done'
#14 14.58     Preparing wheel metadata: started
#14 14.72     Preparing wheel metadata: finished with status 'done'
#14 14.74 Collecting rope==0.22.0
#14 14.76   Downloading rope-0.22.0-py3-none-any.whl (185 kB)
#14 14.81 Collecting s3transfer==0.5.0
#14 14.82   Downloading s3transfer-0.5.0-py3-none-any.whl (79 kB)
#14 14.85 Collecting six==1.16.0
#14 14.88   Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
#14 14.90 Collecting toml==0.10.2
#14 14.91   Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
#14 14.93 Collecting tomli==2.0.0
#14 14.95   Downloading tomli-2.0.0-py3-none-any.whl (12 kB)
#14 14.97 Collecting traitlets==5.0.5
#14 14.99   Downloading traitlets-5.0.5-py3-none-any.whl (100 kB)
#14 15.02 Collecting urllib3==1.26.4
#14 15.04   Downloading urllib3-1.26.4-py2.py3-none-any.whl (153 kB)
#14 15.08 Collecting vulture==2.3
#14 15.09   Downloading vulture-2.3-py2.py3-none-any.whl (25 kB)
#14 15.11 Collecting wcwidth==0.2.5
#14 15.12   Downloading wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
#14 15.15 Collecting werkzeug==2.0.0
#14 15.19   Downloading Werkzeug-2.0.0-py3-none-any.whl (288 kB)
#14 15.28 Collecting wrapt==1.12.1
#14 15.30   Downloading wrapt-1.12.1.tar.gz (27 kB)
#14 15.41 Collecting yamllint==1.26.3
#14 15.43   Downloading yamllint-1.26.3.tar.gz (126 kB)
#14 15.69 Collecting yarl==1.6.3
#14 15.71   Downloading yarl-1.6.3.tar.gz (176 kB)
#14 15.79   Installing build dependencies: started
#14 16.88   Installing build dependencies: finished with status 'done'
#14 16.88   Getting requirements to build wheel: started
#14 16.98   Getting requirements to build wheel: finished with status 'done'
#14 16.98     Preparing wheel metadata: started
#14 17.10     Preparing wheel metadata: finished with status 'done'
#14 17.15 Collecting typing-extensions>=3.6.5
#14 17.15 ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
#14 17.15     typing-extensions>=3.6.5 from https://files.pythonhosted.org/packages/05/e4/baf0031e39cf545f0c9edd5b1a2ea12609b7fcba2d58e118b11753d68cf0/typing_extensions-4.0.1-py3-none-any.whl#sha256=7f001e5ac290a0c0401508864c7ec868be4e701886d5b573a9528ed3973d9d3b (from aiohttp==3.7.4.post0->-r requirements.txt (line 1))
#14 17.25 WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
#14 17.25 You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
------
executor failed running [/bin/sh -c python -m pip install --no-cache-dir -r requirements.txt]: exit code: 1
ERROR: Service 'bot' failed to build : Build failed```

### PR DESCRIPTION
**Pull request by samcat116**
Needed to downgrade our python container to fix a dependency issue with `typing-extensions`. Resolves #180

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,15 +19,15 @@ COPY poetry.lock pyproject.toml /app/
...
-RUN apt-get update && apt-get install gcc=4:10.2.1-1 -y
+FROM python:3.9.9-slim as dev
 WORKDIR /app
-RUN python -m pip install --no-cache-dir -r requirements.txt
+RUN apt-get update && apt-get install gcc=4:10.2.1-1 -y && rm  -rf /var/lib/apt/lists/*
+COPY --from=build /app/requirements-dev.txt /app/requirements.txt
+RUN python -m pip install --no-cache-dir --no-deps -r requirements.txt
 COPY . /app/
-ENTRYPOINT ["python", "-m", "nachomemes.bot", "-d"]
+CMD ["python", "-m", "nachomemes.bot", "-d"]
```
Comment (jswny) [inline Dockerfile]: Why did this change?

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,504 @@
+aiohttp==3.7.4.post0; python_version >= "3.6" and python_full_version >= "3.6.1" and python_full_version < "4.0.0" \
```
Comment (jswny) [inline requirements-dev.txt]: This shouldn't be committed

File: Dockerfile (hunk 1 above)
Comment (samcat116) [reply Dockerfile]: reverted

File: requirements-dev.txt (hunk 2 above)
Comment (samcat116) [reply requirements-dev.txt]: Deleted

**Comment by jswny** [comment]
Python moment.

Good work LGTM!

**Review by jswny — APPROVED** [review]

**Pull request merged**